messaginginapp/src/main/java/io/customer/messaginginapp/ui/bridge/InAppPlatformDelegate.kt

```diff
@@ -0,0 +1,147 @@
+ */
+internal interface InAppPlatformDelegate {
+    fun parseJavaURI(uriString: String): URI
+    fun parseAndroidUri(uriString: String): Uri
+    fun sanitizeUrlQuery(url: String): UrlQuerySanitizer
+    fun decodeBase64ToString(input: String): String
+    fun parsePropertiesFromJson(json: String): Map<String, Any>
+    fun openUrl(url: String)
+    fun openUri(uri: Uri)
+    fun startActivity(intent: Intent)
+
+    /**
+     * Determines if the view should be destroyed when detached from its parent.
+     * The view should be destroyed if the activity is finishing or not changing configurations.
+     * This is useful for managing resources and preventing memory leaks.
+     */
+    fun shouldDestroyViewOnDetach(): Boolean
+
+    /**
+     * Converts the given size from dp to pixels based on the device's screen density.
+     */
+    fun convertDpToPixels(size: Double): Int
+
+    /**
+     * Animates the size of the view to the specified width and height in pixels.
+     * The animation duration can be specified, or a default value will be used.
+     * Callbacks for animation start and end can also be provided.
+     */
+    @UiThread
+    fun animateViewSize(
+        widthInPx: Int? = null,
+        heightInPx: Int? = null,
+        duration: Long? = null,
+        onStart: (() -> Unit)? = null,
+        onEnd: (() -> Unit)? = null
+    )
+}
+
+/**
+ * Default implementation of [InAppPlatformDelegate] that delegates all calls to real Android
+ * framework APIs.
+ * The implementation is coupled to a specific [View] instance, which is used for context and
+ * UI related operations.
+ */
+internal class AndroidInAppPlatformDelegate(
+    private val view: View
+) : InAppPlatformDelegate {
+    private val context: Context
+        get() = view.context
+
+    override fun parseJavaURI(uriString: String): URI {
+        return URI(uriString)
+    }
+
+    override fun parseAndroidUri(uriString: String): Uri {
+        return uriString.toUri()
+    }
+
+    override fun sanitizeUrlQuery(url: String): UrlQuerySanitizer {
+        return UrlQuerySanitizer(url)
+    }
+
+    override fun decodeBase64ToString(input: String): String {
+        val parameterBinary = Base64.decode(input, Base64.DEFAULT)
+        val parameterString = String(parameterBinary, StandardCharsets.UTF_8)
+        return parameterString
+    }
+
+    override fun parsePropertiesFromJson(json: String): Map<String, Any> {
+        val parameterBinary = Base64.decode(json, Base64.DEFAULT)
+        val parameterString = String(parameterBinary, StandardCharsets.UTF_8)
+        val map: Map<String, Any> = HashMap()
+        val properties = Gson().fromJson(parameterString, map.javaClass)
+        return properties
+    }
+
+    override fun openUrl(url: String) {
+        val intent = Intent(Intent.ACTION_VIEW)
+        intent.data = url.toUri()
+        ContextCompat.startActivity(context, intent, null)
+    }
+
+    override fun openUri(uri: Uri) {
```

Comment:
maybe, we can just add the flags boolean here? so that we don't need `openUrl` anymore?